File: src/ast.c
#include "ast.h"
#include <string.h>

AstNode *ast_new(Arena *a, NodeKind kind, SrcLoc loc) {
    AstNode *n = arena_alloc(a, sizeof(AstNode), _Alignof(AstNode));
    memset(n, 0, sizeof(AstNode));
    n->kind = kind;
    n->loc = loc;
    return n;
}

File: kt-src/ast.kt
#include "types.kth"
// ast.kt — AST node constructor

fn ast_new(a: &mut Arena, kind: NodeKind, loc: SrcLoc) -> *AstNode {
    let n: *AstNode = arena_alloc(a, 56, 8)
    memset(n, 0, 56)
    n.kind = kind
    n.loc = loc
    return n
}
